bigquery/tests/system.py

```diff
@@ -819,6 +901,95 @@ def test_sync_query_w_query_params(self):
+                },
+            },
+            {
+                'sql': 'SELECT %s',
+                'expected': False,
+                'query_parameters': [False],
+            },
+            {
+                'sql': 'SELECT %(intval)s',
+                'expected': 123,
+                'query_parameters': {
+                    'intval': 123,
+                },
+            },
+            {
+                'sql': 'SELECT %s',
+                'expected': -123456789,
+                'query_parameters': [-123456789],
+            },
+            {
+                'sql': 'SELECT %(floatval)s',
+                'expected': 1.25,
+                'query_parameters': {
+                    'floatval': 1.25,
+                },
+            },
+            {
+                'sql': 'SELECT LOWER(%(strval)s)',
+                'query_parameters': {
+                    'strval': 'I Am A String',
+                },
+                'expected': 'i am a string',
+            },
+            {
+                'sql': 'SELECT DATE_SUB(%(dateval)s, INTERVAL 1 DAY)',
+                'query_parameters': {
+                    'dateval': datetime.date(2017, 4, 2),
+                },
+                'expected': datetime.date(2017, 4, 1),
+            },
+            {
+                'sql': 'SELECT TIME_ADD(%(timeval)s, INTERVAL 4 SECOND)',
+                'query_parameters': {
+                    'timeval': datetime.time(12, 34, 56),
+                },
+                'expected': datetime.time(12, 35, 0),
+            },
+            {
+                'sql': (
+                    'SELECT DATETIME_ADD(%(datetimeval)s, INTERVAL 53 SECOND)'
+                ),
+                'query_parameters': {
+                    'datetimeval': datetime.datetime(2012, 3, 4, 5, 6, 7),
+                },
+                'expected': datetime.datetime(2012, 3, 4, 5, 7, 0),
+            },
+            {
+                'sql': 'SELECT TIMESTAMP_TRUNC(%(zoned)s, MINUTE)',
+                'query_parameters': {
+                    'zoned': datetime.datetime(
+                        2012, 3, 4, 5, 6, 7, tzinfo=UTC),
+                },
+                'expected': datetime.datetime(2012, 3, 4, 5, 6, 0, tzinfo=UTC),
+            },
+        ]
+        for example in examples:
+            msg = 'sql: {} query_parameters: {}'.format(
+                example['sql'], example['query_parameters'])
+
+            try:
+                Config.CURSOR.execute(
+                    example['sql'], example['query_parameters'])
+            except dbapi.DatabaseError as ex:
+                raise dbapi.DatabaseError('{} {}'.format(ex, msg))
+
+            self.assertEqual(Config.CURSOR.rowcount, 1, msg=msg)
+            row = Config.CURSOR.fetchone()
+            self.assertEqual(len(row), 1, msg=msg)
+            self.assertEqual(row[0], example['expected'], msg=msg)
+            row = Config.CURSOR.fetchone()
+            self.assertIsNone(row, msg=msg)
```

Comment:
Yeah, I added that because I was getting some failures and couldn't tell which example was failing from the message. Seemed like it was good to keep in there.